packages/datadog-plugin-next/test/index.spec.js

```diff
@@ -320,7 +362,215 @@ describe('Plugin', function () {
       })
 
       describe('with configuration', () => {
-        startServer(true)
+        startServer({ withConfig: true, standalone: false, startViaNodeOptions: false }, version)
+
+        it('should execute the hook and validate the status', done => {
+          agent
+            .use(traces => {
+              const spans = traces[0]
+
+              expect(spans[0]).to.have.property('name', 'next.request')
+              expect(spans[0]).to.have.property('service', 'test')
+              expect(spans[0]).to.have.property('type', 'web')
+              expect(spans[0]).to.have.property('resource', 'GET /api/hello/[name]')
+              expect(spans[0]).to.have.property('error', 1)
+              expect(spans[0].meta).to.have.property('span.kind', 'server')
+              expect(spans[0].meta).to.have.property('http.method', 'GET')
+              expect(spans[0].meta).to.have.property('http.status_code', '200')
+              expect(spans[0].meta).to.have.property('foo', 'bar')
+              expect(spans[0].meta).to.have.property('component', 'next')
+            })
+            .then(done)
+            .catch(done)
+
+          axios
+            .get(`http://localhost:${port}/api/hello/world`)
+            .catch(done)
+        })
+      })
+    })
+
+    // Next.js run in standalone
+    // Since it uses `next-server`, only testing automatic instrumentation
+    withVersions('next', 'next', '>=12.0.0 <13.4.0', version => {
```

Comment:
As mentioned in your previous comment, it's possible to filter with semver within `withVersions` instead of repeating which allows also sharing tests more easily.